### DISCUSSION TITLE
Repeated HTML output in Templates

### DISCUSSION THREAD

**Discussion by jason-wilson** [mojolicious/mojo#1774] [Q&A] [answered]
Hi Guys,

Been playing with Mojolicious for some internal applications (and loving it, thanks!) but struggling to find best way to solve a template rendering issue.

Have a html pages that display quite long and complex tables (think a spreadsheet) in one template - this is all fine.

Now have a different template and want to show only matching lines from this original template inside another part of the page (inside a table cell, not that should matter)

Struggling to work out how to generalise this work. In short I want a function/helper/something that takes a perl object (it is a reference to a hash) and render a single html table row that I can use from 2 different templates. Main purpose is not to repeat the templating code.

I can see you could use helpers to call perl functions, but how would I use this to generate the html output?

Sorry if not explaining well enough.

Jason

**Comment by Grinnz** [comment, marked as answer]
Perhaps https://metacpan.org/pod/Mojolicious::Controller#render_to_string

**Reply by jason-wilson** [reply]
How would I call this from inside another template, tried like:
% print render_to_string('table/rule', $rule);

But get error:
Undefined subroutine &Mojo::Template::Sandbox::d5585210f80c68be0f6f5ec9847ed0cc::render_to_string called at template table/list.html.ep line 35,

**Reply by Grinnz** [reply]
It's a controller method, not a helper; from inside the template you can just use include: https://metacpan.org/pod/Mojolicious::Plugin::DefaultHelpers#include

**Reply by jason-wilson** [reply]
Thanks very much that - does just what I needed (and also funding the dumper helper is good to know!)
